saleor/order/utils.py

```diff
@@ -674,6 +674,8 @@ def get_valid_shipping_methods_for_order(
     order: Order,
     shipping_channel_listings: Iterable["ShippingMethodChannelListing"],
     manager: "PluginsManager",
+    *,
+    exclude_methods: bool = True
```

Comment:
Not sure of this. Like we have a helper method that can return us a `valid_shipping_methods` but now we can get `valid_shipping_methods` (as name function) and the methods that are not exactly valid (by setting `exclude_methods=False`). 
Maybe we could extract the first part of the function's body to separate function and call it inside `get_valid_shipping_methods_for_order(`? Then you could use this newly created function and use it in your scenario, then you will not need `exclude_methods` here

Reply:
fixed